Samples.Objc/Samples.ObjC/API/Requests/CreateOrderRequest.m

```diff
@@ -10,35 +10,127 @@
 
 @implementation CreateOrderRequest
 
-- (id)initWith:(OrderIntent)intent purchaseUnits:(NSArray *)purchaseUnits {
+- (id)initWithOrder:(PPCOrderRequest*)order accessToken:(NSString*)token {
   self = [super init];
   if (self) {
-    self.intent = intent;
-    self.purchaseUnits = purchaseUnits;
+    self.order = order;
+    self.accessToken = token;
   }
   return self;
 }
 
 - (NSDictionary *)properties {
+  NSString *intentString = @"";
+  if (self.order.intent == PPCOrderIntentCapture) {
+    intentString = @"CAPTURE";
+  } else {
+    intentString = @"AUTHORIZE";
+  }
 
-  // Convert purchase units to json
   NSMutableArray *purchaseUnitDictionaries = [NSMutableArray new];
-  for (PurchaseUnit *purchaseUnit in [self purchaseUnits]) {
-    NSDictionary *purchaseUnitDictionary = [purchaseUnit dictionaryRepresentation];
+  for (PPCPurchaseUnit *purchaseUnit in self.order.purchaseUnits) {
+    
+    NSMutableArray *itemDictionaries = [NSMutableArray new];
+    
+    if (purchaseUnit.items) {
+      for (PPCPurchaseUnitItem *item in purchaseUnit.items) {
+        
+        NSDictionary *itemUnitAmountDictionary = @{
+          @"currency_code": @"USD",
+          @"value": item.unitAmount.value
+        };
+        
+        NSString *itemCategoryString = nil;
+        if (item.category == PPCPurchaseUnitCategoryDigitalGoods) {
+          itemCategoryString = @"DIGITAL_GOODS";
+        } else {
+          itemCategoryString = @"PHYSICAL_GOODS";
+        }
+        
+        NSDictionary *itemDictionary = @{
+          @"name": item.name,
+          @"unit_amount": itemUnitAmountDictionary,
+          @"quantity": item.quantity,
+          @"category": itemCategoryString
+        };
+        
+        [itemDictionaries addObject:itemDictionary];
```

Comment:
For clarity's sake, would it make sense to abstract these blocks to their own functions? so the interface would be like:

```objc
if (purchaseUnit.items) {
  [self addValuesFromItems: purchasUnit.items toArray: itemDictionaries];
}
```

What do you think? might be easier to follow what exactly is being evaluated for and added

Reply:
I think it makes a lot of sense!!!
I think as a future enhancement we can add custom ObjC encoder for `OrderRequest` in the SDK like you mentioned before so merchants won't need to do all these!!! :smile: